File: CS3315/Assignments/Week3Assignment/scripts/topTenHuckFinn.rb
#!/usr/bin/env ruby

# this is our regular expression
RE = /[[:alpha:]']+/

# open("HuckFinn.txt") => open the file
# read.downcase        => convert the contents to lowercase
# scan(RE)             => find all substrings that match the regular expression
# tally                => count how many times each substring appears
# max_by(10, &:last)   => only keep the top ten based on their tallies
count = open("HuckFinn.txt").read.downcase.scan(RE).tally.max_by(10, &:last)

# display the top ten
count.each{|ar| puts ("   " + ar[1].to_s + " " + ar[0].to_s)}

File: CS3315/Assignments/Week3Assignment/scripts/topTenHuckFinn.py
#!/usr/bin/env python3

# this is needed in order to use regular expressions
import re

f = open("HuckFinn.txt", errors='ignore')

# read the file into memory.
# lines is now an array full of strings
# that are each line from HuckFinn.txt
lines = f.readlines()

# close the file since its contents are now in memory
f.close()

# this dictionary will have each word as a key, and
# the number of times that it appears as the value
data = {}

for line in lines:
    # convert the line to lowercase
    line = line.lower()
    
    # find all substrings that match the regular expression
    words = re.findall("[a-z']+", line)
    
    # this for loop will count the number of times each
    # substring has occured
    for word in words:
        if (word in data):
            data[word] += 1
        else:
            data[word] = 1
            
# sort the key, value pairs based on their values
top_ten = sorted(data, key=data.get, reverse=True)[:10]

# display the top ten
for key in top_ten:
    print("  ", data[key], key)
